doc/modules/gaussian_process.rst

```diff
@@ -598,6 +598,25 @@ shown in the following figure:
    :target: ../auto_examples/gaussian_process/plot_gpr_prior_posterior.html
    :align: center
 
+Custom kernel
+-------------
+
+If none of the above kernels fit your needs, you have the option of creating a custom kernel. A custom kernel must include the following methods:
```

Comment:
Not really... in practice most implementations will inherit from `Kernel` which provides (default) implementations of these things.

What needs to be written for a custom kernel is `__call__`, `diag`, `hyperparameters` or `hyperparameter_xyz`., etc.

You should try to give an example of some simple, contrived kernel with a parameter and a hyperparameter.

Reply:
Hi @jnothman, are `StationaryKernelMixin` and `NormalizedKernelMixin` classes already implementing `diag` and `is_stationary` methods that could be used in principle by every possible customised Kernel (nothing of particularly exotic is coming in my mind)? If this is the case, should be inheriting from these classes be enough? If yes, just `__called__` should be implemented (mine is more a question than an assertion :) ).